### ISSUE TITLE
fzf-lua.defaults is nil

### ISSUE THREAD

**Issue by wjdwndud0114** [ibhagwan/fzf-lua#993]
`require('fzf-lua').defaults` is nil.

Going back to commit https://github.com/ibhagwan/fzf-lua/commit/6991c1f86eb9fa449a186ab047e88063d3a14484 brings it back

**Comment by ibhagwan** [comment]
This commit you identified as breaking has been around for a while (2 weeks) and there have been no major complaints so I'm assuming this isn't a widespread issue or there would've been countless of issues about this.

Nevertheless I would very much like to solve this but you have provided basically zero information on your setup, can you please post your `setup` function (as requested in the template...) and also try with `mini.sh` and let me know if that works (also requested in the issue template  so we can eliminate neovim config issues)?

Basically refill this issue with the template and I'll solve it, the below info is requested **for a reaoson**:

> ### Info
> - Operating System:
> - Shell:
> - Terminal:
> - `nvim --version`:
> - `fzf --version`:

> Check if your problem is reproducible with minimal init, by running:
> sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"

**Comment by ibhagwan** [comment]
> require('fzf-lua').defaults

This is also `nil` in my setup but this isn't required for fzf-lua to operate, that's because the commit removed defaults from the exported modules.

I'm assuming you have been using this for your own custmoization?

You can still acess the deafaults via `require('fzf-lua.defaults')`

**Comment by wjdwndud0114** [comment]
Thanks for addressing this!

Sorry, I thought the setup was irrelevant here.

Yes I have a custom files search to make it faster with icons:
```lua
  -- speed up icon: https://github.com/MaartenStaa/file-web-devicons
  local fzf = require('fzf-lua')

  vim.keymap.set('n', '<C-p>', function()
    fzf.fzf_exec('fd -H --type f --strip-cwd-prefix | ~/.dotfiles/file-web-devicon', {
      actions = fzf.defaults.actions.files,
      fzf_opts = { ['--nth'] = 2, ['--delimiter'] = fzf.utils.nbsp },
      previewer = 'builtin',
    })
  end, { noremap = true })
```

> @wjdwndud0114 - [a56c006](https://github.com/ibhagwan/fzf-lua/commit/a56c006140e338a4fa9e361d91d7d30ec7bada41) should restore the defaults module export.

This works but the export is a table of { globals, defaults } so I have to do require('fzf-lua').defaults.defaults which is fine but could be confusing

**Comment by ibhagwan** [comment]
> Sorry, I thought the setup was irrelevant here.

Correct it wasn’t relevant, but you haven’t provided other information and it took me a minute to understand what you meant :)

**Comment by ibhagwan** [comment]
> This works but the export is a table of { globals, defaults } so I have to do require('fzf-lua').defaults.defaults which is fine but could be confusing and would require some updates to the docs

I see what you mean, I’ll fix it.

**Comment by ibhagwan** [comment]
@wjdwndud0114, https://github.com/ibhagwan/fzf-lua/commit/a614aee1ed03eddaadcc9f902c8a6fb929ba2903 this should take care of it, definitely makes more sense now.

**Comment by wjdwndud0114** [comment]
Nice, thanks a lot! Amazing plugin :)

**Comment by ibhagwan** [comment]
> Nice, thanks a lot! Amazing plugin :)

I had to revert this temporarily to a different branch as this was causing icons to be hidden, I'll have more time later to test it and merge properly, for the time being the commit is in a different branch:
https://github.com/ibhagwan/fzf-lua/tree/defaults_refactor

**Comment by rafikdraoui** [comment]
For the record, it did break my setup as well :slightly_smiling_face:  I was relying on `require("fzf-lua").defaults.keymap` to get a copy of the default key mappings in order to extend them.

I was planning on opening an issue about it, but I wanted to do some research about my use case of extending default keymaps before doing so, in case I was missing something obvious about it (and avoid a [XY problem](https://xyproblem.info/)).

I didn't have any time to put to it yet, so instead I just temporarily worked around it by using `require("fzf-lua.defaults").defaults.keymap`, and putting it off for a later day :smile:

**Comment by ibhagwan** [comment]
@rafikdraoui I’ll fix it today :)

**Comment by ibhagwan** [comment]
@wjdwndud0114 , @rafikdraoui - use this commit https://github.com/ibhagwan/fzf-lua/commit/2cd8e0c32fcdad48d8d56120c559d15a10affb3f.
